grid/address/geocoding/maps/here.suma

```diff
@@ -0,0 +1,144 @@
+profile = "address/geocoding@3.1"
+provider = "here"
+
+// Here API documentation
+//   https://developer.here.com/documentation
+
+map Geocode {
+    payload = input.query || `${input.streetAddress || ""} ${input.addressLocality || ""} ${input.addressRegion || ""} ${input.postalCode || ""} ${input.addressCountry || ""}`;
+
+    http GET "geocode" "/v1/geocode" {
+        security "apiKey"
+        
+        request {
+            query {
+                q = payload
+            }
+        }
+
+        response 200 "application/json" {
+            return map error if (body.items.length === 0) {
+                title = 'Results not found',
+                detail = 'No results were found for specified address'
+            }
+            
+            map result {
+                latitude = body.items[0].position.lat
+                longitude = body.items[0].position.lng
+            }
+        }
+
+        response 400 {
```

Comment:
We are mapping title and cause for all status codes the same way I suggest to DRY the code and handle errors in operation.

```
operation MapHereError {
  statusCode = args.statusCode
  body = args.body

  return if (statusCode === 400 ||
             statusCode === 401,
             ...) {
    title = body.title
    detail = body.cause
  }

  return {
    title = "Unknown error"
    detail = (`Unknown error occurred. Status: ${statusCode}. Provider detail: ${body.title} ${body.cause}`)
  }
}
```

Reply:
Also error handling is the same for both use-cases

Reply:
Good point 👍🏻